src/bidiMapper/utils/cdpServer.spec.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2021 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { connectCdp } from './cdpClient';
+import { StubServer } from './tests/stubServer.spec';
+import { assert as chai_assert } from 'chai';
+import { assert as sinon_assert } from 'sinon';
+import { Protocol } from 'devtools-protocol';
+
+describe('CdpClient tests.', async () => {
+  it(`given CdpClient, when some command is called, then cdpBindings should be
+      called with proper values`, async () => {
+    const expectedMessageStr = JSON.stringify({
+      id: 0,
+      method: 'Target.activateTarget',
+      params: {
+        targetId: 'TargetID',
+      },
+    });
+
+    const mockCdpServer = new StubServer();
+
+    const cdpClient = connectCdp(mockCdpServer);
+    cdpClient.Target.activateTarget({ targetId: 'TargetID' });
+
+    sinon_assert.calledOnceWithExactly(
+      mockCdpServer.sendMessage,
+      expectedMessageStr
+    );
+  });
+
+  it(`given some command is called, when CDP command is done, then
+        'sendMessage' promise is resolved`, async () => {
+    const mockCdpServer = new StubServer();
+    const cdpClient = connectCdp(mockCdpServer);
+
+    // Get handler 'onMessage' to notify 'cdpClient' about new CDP messages.
+    const onMessage = mockCdpServer.getOnMessage();
+
+    // Send CDP command and store result promise.
+    const commandPromise = cdpClient.Target.activateTarget({
+      targetId: 'TargetID',
+    });
+
+    // Verify CDP command was sent.
+    sinon_assert.calledOnce(mockCdpServer.sendMessage);
+
+    // Notify 'cdpClient' the CDP command is finished.
+    onMessage(JSON.stringify({ id: 0, result: {} }));
+
+    // Assert 'cdpClient' resolved message promise.
+    await commandPromise;
```

Comment:
Is there nothing to assert about what this resolves to?

Reply:
there is another test verifies the resolve result